admin/admin.py

```diff
@@ -186,29 +186,24 @@ def _hide_file_upload(self, form):
         return file
 
     def _handle_file_upload(self, file, form, model):
-        got_new_file = False
         if file.data:
             if model.has_file and os.path.exists(document_path(model.id)) and not app.config['DEBUG']:
                 # delete old file
                 os.unlink(document_path(model.id))
             save_file(model, file.data)
             if model.validated:
-                got_new_file = True
-
-        if model.validation_time is None and model.validated:
-            # document has just been validated for the first time
-            model.validation_time = datetime.datetime.now()
-            got_new_file = True
-
-        if got_new_file:
-            if model.document_type != 'written':
-                barcode.bake_barcode(model)
-            config.document_validated(document_path(model.id))
+                if model.document_type != 'written':
+                    barcode.bake_barcode(model)
```

Comment:
This way you only bake barcodes into PDFs if new files are uploaded -- student-submitted PDFs won't be given a barcode upon validation